### PR TITLE
[ATSCALE-18321]- Pull collibra Assets and Attributes

### PR DESCRIPTION
**Pull request by pepcus-mukatip**
**Tickets**

- https://atscale.atlassian.net/browse/ATSCALE-18321

**Description**

- Fetch assets from Collibra.
- Retrieve attributes from Collibra.
- Map assets with their corresponding attributes.

**Testing** 

- Testing done through postman
(Note: we are using wrapper API, and the API response is utilized in other API calls. so unable to attach screenshot until the other API integration done)

### CODE REVIEW

File: src/main/java/com/collibra/marketplace/atscale/controller/EntryPointController.java [hunk 1]
```diff
@@ -69,4 +69,10 @@ public JsonNode rmObsTriggeredByApiRequest() {
 		LOGGER.info("Remove obsolete assets triggered via API request");
 		return this.mainProcessor.removeObsolete();
 	}
+	@PostMapping("/syncCollibraToAtscale")
```
Comment (tanmay-atscale) [inline src/main/java/com/collibra/marketplace/atscale/controller/EntryPointController.java:72]: /syncCollibraToAtscale : /syncollibratoatscale or /sync-collibra-atscale

File: src/main/java/com/collibra/marketplace/atscale/model/Attribute.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
+package com.collibra.marketplace.atscale.model;
+
+import lombok.Data;
+
+@Data
+public class Attribute {
```
Comment (tanmay-atscale) [inline src/main/java/com/collibra/marketplace/atscale/model/Attribute.java:6]: if it is specific to Measure then MeasureAttribute. Okay if it is generic.

File: src/main/java/com/collibra/marketplace/atscale/model/Attribute.java [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    private String Id;
+    private String name;
+    private String typeId;
+    private String AssetId;
```
Comment (tanmay-atscale) [inline src/main/java/com/collibra/marketplace/atscale/model/Attribute.java:10]: assetId

File: src/main/java/com/collibra/marketplace/atscale/model/Attribute.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package com.collibra.marketplace.atscale.model;
+
+import lombok.Data;
+
+@Data
+public class Attribute {
+    private String Id;
```
Comment (tanmay-atscale) [inline src/main/java/com/collibra/marketplace/atscale/model/Attribute.java:7]: id

File: src/main/java/com/collibra/marketplace/atscale/model/Attribute.java (hunk 2 above)
Comment (pepcus-mukatip) [reply src/main/java/com/collibra/marketplace/atscale/model/Attribute.java:6]: It is generic.

File: src/main/java/com/collibra/marketplace/atscale/model/Measure.java [hunk 5]
```diff
@@ -46,6 +47,7 @@ public class Measure {
   private boolean isMetricalAttribute;
   private String parentLevelId;
   private String parentLevelName;
+  private List<Attribute> attributeList;
 
   private static final Logger LOGGER = LoggerFactory.getLogger(Measure.class);
```
Comment (tanmay-atscale) [inline src/main/java/com/collibra/marketplace/atscale/model/Measure.java:52]: why we need logger in model?

File: src/main/java/com/collibra/marketplace/atscale/service/MainProcessor.java [hunk 6]
```diff
@@ -756,4 +757,66 @@ public boolean domainExists(CollibraDomain domain) {
...
+        for (Map.Entry<String, Project> currentProjectPair : projectsMap.entrySet()) {
+            List<Measure> collibraProjectMeasureList = getCollibraProjectMeasureList(collibraProjectList, currentProjectPair.getKey());
+            Map<String, Project> currentProject = new HashMap<>();
+            currentProject.put(currentProjectPair.getKey(), currentProjectPair.getValue());
+            List<Measure> allMeasures = retrieveMeasures(currentProject);
+            if (!Objects.requireNonNull(collibraProjectMeasureList).isEmpty()) {
+                Map<String, Measure> collibraMeasureMap = convertCollibraMeasureASMap(collibraProjectMeasureList);
+                boolean flag = compareMeasureDescription(collibraMeasureMap, allMeasures);
+                if (flag) {
+                   // Call API to publish project
```
Comment (tanmay-atscale) [inline src/main/java/com/collibra/marketplace/atscale/service/MainProcessor.java:775]: you can add  //TODO

**Review by tanmay-atscale — COMMENTED** [review]
Please check the comments...

File: src/main/java/com/collibra/marketplace/atscale/controller/EntryPointController.java (hunk 1 above)
Comment (pepcus-mukatip) [reply src/main/java/com/collibra/marketplace/atscale/controller/EntryPointController.java]: done with  /sync-collibra-atscale

File: src/main/java/com/collibra/marketplace/atscale/model/Attribute.java (hunk 3 above)
Comment (pepcus-mukatip) [reply src/main/java/com/collibra/marketplace/atscale/model/Attribute.java]: done

File: src/main/java/com/collibra/marketplace/atscale/model/Attribute.java (hunk 4 above)
Comment (pepcus-mukatip) [reply src/main/java/com/collibra/marketplace/atscale/model/Attribute.java]: done

File: src/main/java/com/collibra/marketplace/atscale/service/MainProcessor.java (hunk 6 above)
Comment (pepcus-mukatip) [reply src/main/java/com/collibra/marketplace/atscale/service/MainProcessor.java]: done

File: src/main/java/com/collibra/marketplace/atscale/model/Measure.java (hunk 5 above)
Comment (pepcus-mukatip) [reply src/main/java/com/collibra/marketplace/atscale/model/Measure.java:52]: The statement already exists and has not been added in this PR.

**Review by tanmay-atscale — APPROVED** [review]
LGTM

**Review by pepcus-bhawani — APPROVED** [review]

**Pull request merged**